core/math/quaternion.h

```diff
@@ -141,16 +141,26 @@ struct [[nodiscard]] Quaternion {
 	}
 
 	Quaternion(const Vector3 &p_v0, const Vector3 &p_v1) { // Shortest arc.
-		Vector3 c = p_v0.cross(p_v1);
+#ifdef MATH_CHECKS
+		ERR_FAIL_COND_MSG(p_v0.is_zero_approx() || p_v1.is_zero_approx(), "The vectors must not be zero.");
+#endif
+		Vector3 n0 = p_v0.normalized();
+		Vector3 n1 = p_v1.normalized();
+
+		real_t d = n0.dot(n1);
+		real_t n = 1.0f - (real_t)CMP_EPSILON;
+		if (d > n) {
```

Comment:
Should this be `<`?

Reply:
`n0.dot(n1) > 1.0f - (real_t)CMP_EPSILON` is correct because it returns no rotation if the dot product is 1.